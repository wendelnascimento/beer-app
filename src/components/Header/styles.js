import styled from 'styled-components';

export const Wrapper = styled.header`
  max-width: 100vw;
  background: #fff;
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid #fafafa;
`;

export const Title = styled.h1`
  font-weight: 300;
  margin: 0;
`;
