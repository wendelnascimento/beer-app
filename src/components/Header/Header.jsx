import React from 'react';

import { Wrapper, Title } from './styles';

const Header = () => (
  <Wrapper>
    <Title>
      Beer App
      <span role="img" aria-label="Beer">
        🍺
      </span>
    </Title>
  </Wrapper>
);

export default Header;
