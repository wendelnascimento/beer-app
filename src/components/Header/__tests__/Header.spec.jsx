import React from 'react';
import { shallow, mount } from 'enzyme';

import Header from '../Header';

describe('<Header />', () => {
  it('does render properly', () => {
    shallow(<Header />);
  });

  it('matches stored snapshot', () => {
    const wrapper = mount(<Header />);

    expect(wrapper).toMatchSnapshot();
  });

  it('does have Beer App title', () => {
    const wrapper = mount(<Header />);

    expect(wrapper.find('h1').text()).toBe('Beer Appüç∫');
  });
});
