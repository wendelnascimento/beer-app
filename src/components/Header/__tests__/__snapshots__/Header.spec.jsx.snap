// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches stored snapshot 1`] = `
.c0 {
  max-width: 100vw;
  background: #fff;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-bottom: 1px solid #fafafa;
}

.c1 {
  font-weight: 300;
  margin: 0;
}

<Header>
  <styled.header>
    <header
      className="c0"
    >
      <styled.h1>
        <h1
          className="c1"
        >
          Beer App
          <span
            aria-label="Beer"
            role="img"
          >
            üç∫
          </span>
        </h1>
      </styled.h1>
    </header>
  </styled.header>
</Header>
`;
